### PR TITLE
Use sof_da7219 to support max98390 on CML platform

### PR DESCRIPTION
**Pull request by brentlu**
Currently we use bxt_da7219_max98357a machine driver to support following devices:
- da7219+max98357a on APL (SST)
- da7219+max98357a on GLK (SOF)
- da7219+max98357a on CML (SOF)
- da7219+max98390 on CML (SOF)

This PR will add the support of da7219+max98390 on CML to sof_da7219 machine driver and remove the corresponding code in bxt_da7219_max98357a.

Code tested on Samsung CML Chromebook.

### CODE REVIEW

File: sound/soc/intel/boards/Kconfig [hunk 1]
```diff
@@ -313,14 +312,14 @@ config SND_SOC_INTEL_BXT_DA7219_MAX98357A_COMMON
 if SND_SOC_INTEL_APL
 
 config SND_SOC_INTEL_BXT_DA7219_MAX98357A_MACH
-	tristate "Broxton with DA7219 and MAX98357A/MAX98390 in I2S Mode"
+	tristate "Broxton with DA7219 and MAX98357A in I2S Mode"
```
Comment (plbossart) [inline sound/soc/intel/boards/Kconfig:315]: same problem as last time, if a user does a 'make olddefconfig' then they may not enable the 'new' config and their sound support would be broken. For backwards compatibility using 'imply' seems required, no?

Comment (brentlu) [reply sound/soc/intel/boards/Kconfig:315]: Sorry I forgot this. Added to SND_SOC_INTEL_CML_LP_DA7219_MAX98357A_MACH config.

File: sound/soc/intel/boards/sof_da7219.c [hunk 2]
```diff
@@ -281,6 +281,9 @@ sof_card_dai_links_create(struct device *dev, struct snd_soc_card *card,
 			return -EINVAL;
 		}
 		break;
+	case CODEC_MAX98390:
+		max_98390_dai_link(dev, ctx->amp_link);
```
Comment (bardliao) [inline sound/soc/intel/boards/sof_da7219.c:286]: We need to select SND_SOC_MAX98390 in config SND_SOC_INTEL_SOF_DA7219_MACH if sof_da7219 supports max98390.

Comment (brentlu) [reply sound/soc/intel/boards/sof_da7219.c:286]: fixed. thanks for reminding.

**Review by bardliao — APPROVED** [review]

**Review by plbossart — APPROVED** [review]

**Pull request merged**